hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -281,12 +331,40 @@ private boolean checkKeyReclaimable(
         return false;
       }
 
-      //TODO: [SNAPSHOT] Handle Renamed Keys
-      String dbKey = ozoneManager.getMetadataManager()
-          .getOzoneKey(deletedKeyInfo.getVolumeName(),
-              deletedKeyInfo.getBucketName(), deletedKeyInfo.getKeyName());
+      if (bucketInfo.getBucketLayout().isFileSystemOptimized()) {
+        // Handle FSO buckets
+        dbKey = ozoneManager.getMetadataManager().getOzonePathKey(volumeId,
+            bucketInfo.getObjectID(), deletedKeyInfo.getParentObjectID(),
+            deletedKeyInfo.getKeyName());
+      } else {
+        dbKey = ozoneManager.getMetadataManager()
+            .getOzoneKey(deletedKeyInfo.getVolumeName(),
+                deletedKeyInfo.getBucketName(), deletedKeyInfo.getKeyName());
+      }
+
+      // renamedKeyTable: volumeName/bucketName/objectID -> OMRenameKeyInfo
+      String dbRenameKey = ozoneManager.getMetadataManager().getRenameKey(
+          deletedKeyInfo.getVolumeName(), deletedKeyInfo.getBucketName(),
+          deletedKeyInfo.getObjectID());
+
+      OmKeyRenameInfo renamedKeyInfo = renamedKeyTable.getIfExist(dbRenameKey);
+
+      boolean isKeyRenamed = false;
+      String dbOriginalKey = null;
+      // Condition: key should not exist in renamedKeyTable of the current
+      // snapshot and keyTable of the previous snapshot.
+      // Check key exists in renamedKeyTable of the Snapshot
+      if (renamedKeyInfo != null && !renamedKeyInfo
+          .getOmKeyRenameInfoList().isEmpty()) {
```

Comment:
We can refactor renamedKeyTable to keep a single `String` in its value. Can be done in HDDS-8189 as well.

Reply:
Sure will update it in that JIRA.

Reply:
Refactored the code change in #4436